frontend/js/src/settings/delete-listens/DeleteListens.tsx

```diff
@@ -43,27 +24,42 @@ export default function DeleteListens() {
           "Content-Type": "application/json",
         },
       });
+      if (!response.ok) {
+        const errJson = await response.json();
+        throw errJson.message ?? "Error";
+      }
 
       toast.success(
         <ToastMsg
           title="Success"
-          message="Your listens have been enqueued for deletion."
+          message="Your listens have been enqueued for deletion. Please note your listens will still be visible for up to an hour."
         />
       );
 
-      // TODO: Should be replaced by redirect using react-router-dom
       setTimeout(() => {
-        window.location.href = `/user/${name}/`;
+        navigate(`/user/${name}/`);
       }, 3000);
     } catch (error) {
       toast.error(
         <ToastMsg
           title="Error"
-          message={`Error while deleting listens for user ${name}`}
+          message={
+            <>
+              Error while deleting listens for user {name}:{error.toString()}
```

Comment:
Just a minor thing which caught my eye while briefly skimming this PR:
```suggestion
              Error while deleting listens for user {name}: {error.toString()}
```